test/StrikeSelectionE2E.ts

```diff
@@ -0,0 +1,250 @@
+      wethPriceOracleAddress,
+      usdcPriceOracleAddress
+    );
+
+    strikeSelection = await StrikeSelection.deploy(
+      optionsPremiumPricer.address,
+      1000,
+      100
+    );
+
+    wethPriceOracle = await ethers.getContractAt(
+      "IPriceOracle",
+      await optionsPremiumPricer.priceOracle()
+    );
+  });
+
+  describe("setDelta", () => {
+    time.revertToSnapshotAfterEach();
+
+    it("reverts when not owner call", async function () {
+      await expect(
+        strikeSelection.connect(signer2).setDelta(5000)
+      ).to.be.revertedWith("Ownable: caller is not the owner");
+    });
+
+    it("sets the delta", async function () {
+      await strikeSelection.connect(signer).setDelta(5000);
+      assert.equal((await strikeSelection.delta()).toString(), "5000");
+    });
+  });
+
+  describe("setStep", () => {
+    time.revertToSnapshotAfterEach();
+
+    it("reverts when not owner call", async function () {
+      await expect(
+        strikeSelection.connect(signer2).setStep(50)
+      ).to.be.revertedWith("Ownable: caller is not the owner");
+    });
+
+    it("sets the step", async function () {
+      await strikeSelection.connect(signer).setStep(50);
+      assert.equal(
+        (await strikeSelection.step()).toString(),
+        BigNumber.from(50)
+          .mul(BigNumber.from(10).pow(await wethPriceOracle.decimals()))
+          .toString()
+      );
+    });
+  });
+
+  describe("getStrikePrice", () => {
+    time.revertToSnapshotAfterEach();
+
+    let underlyingPrice: BigNumber;
+    let deltaAtUnderlying: BigNumber;
+    let expiryTimestamp: BigNumber;
+
+    beforeEach(async () => {
+      await updateVol();
+      underlyingPrice = await optionsPremiumPricer.getUnderlyingPrice();
+      underlyingPrice = underlyingPrice.sub(
+        BigNumber.from(underlyingPrice).mod(await strikeSelection.step())
+      );
+      expiryTimestamp = (await time.now()).add(WEEK);
+      deltaAtUnderlying = await optionsPremiumPricer.getOptionDelta(
+        underlyingPrice,
+        expiryTimestamp
+      );
+    });
+
+    it("reverts on timestamp being in the past", async function () {
+      const isPut = false;
+      await expect(
+        strikeSelection.getStrikePrice((await time.now()).sub(100), isPut)
+      ).to.be.revertedWith("Expiry must be in the future!");
+    });
+
+    it("gets the correct strike price given delta for calls", async function () {
+      const isPut = false;
+      const targetDelta = await strikeSelection.delta();
+      const [strikePrice, delta] = await strikeSelection.getStrikePrice(
+        expiryTimestamp,
+        isPut
+      );
+
+      assert.equal(
+        strikePrice.toString(),
+        underlyingPrice
+          .add(
+            deltaAtUnderlying
+              .sub(targetDelta)
+              .div(500)
+              .mul(100)
+              .mul(BigNumber.from(10).pow(await wethPriceOracle.decimals()))
+          )
+          .toString()
+      );
```

Comment:
Do you mind helping clarify what this snippet does? Why do we need to `.div(500).mul(100)`

Reply:
Okay I see now. You're just calculating the final strike price given the delta at underlying. Why don't we just hardcode the results in the test itself?

Reply:
we could do that. but just in case we wanted to customize the target delta within specific test we can do that (instead of once in beforeEach).

the .div(500) is because I found that the strike price moves by 100 (for ether) at approximately blocks of 0.05 in delta.